src/shared/components/oncoprint/controls/OncoprintControls.tsx

```diff
@@ -273,6 +307,45 @@ export default class OncoprintControls extends React.Component<
     private onHeatmapProfileSelect(option: { label: string; value: string }) {
         this.props.handlers.onSelectHeatmapProfile &&
             this.props.handlers.onSelectHeatmapProfile(option.value);
+        const genericAssayType =
+            this.props.molecularProfileIdToMolecularProfile &&
+            this.props.molecularProfileIdToMolecularProfile[option.value]
+                ? this.props.molecularProfileIdToMolecularProfile[option.value]
+                      .genericAssayType
+                : undefined;
+
+        if (
+            this.props
+                .selectedGenericAssayEntitiesGroupByGenericAssayTypeFromUrl &&
+            genericAssayType &&
+            this.props
+                .selectedGenericAssayEntitiesGroupByGenericAssayTypeFromUrl[
+                genericAssayType
+            ] &&
+            this.props.state.selectedHeatmapProfileId &&
+            this.props
+                .genericAssayEntitiesSelectOptionsGroupByMolecularProfileId &&
+            this.props
+                .genericAssayEntitiesSelectOptionsGroupByMolecularProfileId[
+                this.props.state.selectedHeatmapProfileId
+            ]
```

Comment:
can you compute this in a separate line/function and give it a name so its easier to follow whats happening

Reply:
Sure.